dataclients/kubernetes/redirect.go

```diff
@@ -87,16 +88,11 @@ func initRedirectRoute(r *eskip.Route, code int) {
 	}
 	r.Headers[forwardedProtoHeader] = "http"
 
-	// the below duplicate any-path (.*) is set to make sure that
-	// the redirect route has a higher priority during matching than
-	// the normal routes that may have max 2 predicates: path regexp
-	// and host.
-	//
-	// A better solution might be to implement a Weight() predicate
-	// and apply it here.
-	//
-	r.PathRegexps = append(r.PathRegexps, ".*")
-	r.PathRegexps = append(r.PathRegexps, ".*")
+	// Give this route a higher weight so that it will get precedence over existing routes
+	r.Predicates = append([]*eskip.Predicate{{
+		Name: routing.WeightPredicateName,
+		Args: []interface{}{float64(1000)},
```

Comment:
float64 appears in many places, because some time ago we decided that we parse numbers always as float. While this should be revisited at some point, there are already filters and predicates that also accept numbers as ints, and if we define in-memory eskip objects, and the filter or predicate accepts ints, then it's unnecessary to pass in floats. The weight predicates accepts ints, so let's just use an int here, too.

Reply:
If I don't add the `float64` it will become an `int`, and then a lot of tests start to fail. Example:
```
    TestEnableHTTPSRedirectFromIngress: httpsredirect_test.go:372:   []*eskip.Route{
          	&{Id: "kube___catchall__api_example_org____", HostRegexps: []string{"^api[.]example[.]org$"}, BackendType: s"shunt"},
          	&{Id: "kube___catchall__www_example_org____", HostRegexps: []string{"^www[.]example[.]org$"}, BackendType: s"shunt"},
          	&{
          		... // 5 identical fields
          		Headers:       map[string]string{"X-Forwarded-Proto": "http"},
          		HeaderRegexps: nil,
          		Predicates: []*eskip.Predicate{
          			&{
          				Name: "Weight",
        - 				Args: []interface{}{int(1000)},
        + 				Args: []interface{}{float64(1000)},
          			},
          		},
```

How to fix?

Reply:
my bad, didn't realize that in the tests we compare the routing output to routing parsed from text. :+1: